algorithm.go

```diff
@@ -48,6 +51,13 @@ const (
 // RFC 8152 16.4: https://datatracker.ietf.org/doc/html/rfc8152#section-16.4
 type Algorithm int64
 
+// AlgorithmFromString returns the Algorithm corresponding to the specified
+// name.
+// NOTE: there are currently no registered string values for an algorithm.
+func AlgorithmFromString(v string) Algorithm {
+	return AlgorithmInvalid
```

Comment:
Can we remove this function?

Reply:
It's there to support string marshalling for Algorithm. I can revert that entirely, but the spec does say that an alogrithm can  be a string (though there are no string values currently registred with IANA). Allowing it be a string but then reporting the value as invalid seems more correct than disallowing strings altogether.

Reply:
But this function can't be overridden by downstream users, so it will always return `AlgorithmFromString`. I don't see the point on exposing it publicly.

Reply:
Good point. I'll make it private.